barretenberg/cpp/src/barretenberg/commitment_schemes/gemini/gemini_impl.hpp

```diff
@@ -308,18 +279,57 @@ std::vector<typename GeminiProver_<Curve>::Claim> GeminiProver_<Curve>::compute_
         }
     }
 
-    std::vector<Claim> opening_claims;
-    opening_claims.reserve(num_variables + 1);
+    return { std::move(A_0_pos), std::move(A_0_neg) };
+};
 
-    // Compute first opening pair {r, A₀(r)}
-    Fr evaluation = fold_polynomials[0].evaluate(r_challenge);
-    opening_claims.emplace_back(Claim{ fold_polynomials[0], { r_challenge, evaluation } });
-    // Compute the remaining m opening pairs {−r^{2ˡ}, Aₗ(−r^{2ˡ})}, l = 0, ..., m-1.
-    for (size_t l = 0; l < num_variables; ++l) {
-        evaluation = fold_polynomials[l + 1].evaluate(-r_squares[l]);
-        opening_claims.emplace_back(Claim{ fold_polynomials[l + 1], { -r_squares[l], evaluation } });
```

Comment:
This line was resulting in a copy of each fold polynomial. Should have used `std::move()`